tcpinfo.proto

```diff
@@ -0,0 +1,266 @@
+// Up to date as of net-next e147161b1aa037a7150dc7c69d77f16ba6001717
+
+// These protos should ONLY contain raw data from tcpinfo collection.
+// Meta-data, such as experiment name or collection time, and any data from
+// sources other than tcpinfo should be handles OUTSIDE these protos.
+
+syntax = "proto2";
+
+package mlab.tcp_info;
+
+/*
+   This is a work in progress.
+   Generally, this reflects the kernel's struct tcp_info.  However, we have
+   some flexibility to simplify some bits, largely from protobuf's has_xxx
+   functionality.  For example, some of the OPT_XXX flags in the options field
+   can be elided in this protobuf, because their values are reflected in
+   other fields, such as wscale fields.
+
+We should consider collecting any or all of the extensions identified in
+inet_diag.h Extensions:
+enum {
+        INET_DIAG_NONE,
+        INET_DIAG_MEMINFO,  // Done  (subset of skmeminfo)
+        INET_DIAG_INFO,     // Done
+        INET_DIAG_VEGASINFO,  // Done
+        INET_DIAG_CONG,  // Done - string field in top level message.
+        INET_DIAG_TOS,
+        INET_DIAG_TCLASS,
+        INET_DIAG_SKMEMINFO,  // Done
+        INET_DIAG_SHUTDOWN,
+        INET_DIAG_DCTCPINFO,  // Done
+        INET_DIAG_PROTOCOL,  // Done - string in top level message.
+        INET_DIAG_SKV6ONLY,
+        INET_DIAG_LOCALS,
+        INET_DIAG_PEERS,
+        INET_DIAG_PAD,
+        INET_DIAG_MARK,   // NEW - should we include this???
+        INET_DIAG_BBRINFO,  // NEW Done.
+        __INET_DIAG_MAX,
+};
+
+*/
+
+message EndPoint {
+  optional int32 port = 1;
+  optional bytes ip = 2;  // 4 or 16 bytes for ipv4 or ipv6 address.
+}
+
+message InetSocketIDProto {
+  optional EndPoint source = 1;
+  optional EndPoint destination = 2;
+  optional int32 interface = 5;
+  optional fixed64 cookie = 6;
+}
+
+message  InetDiagMsgProto {
+  enum AddressFamily {
+    AF_UNSPEC = 0;
+    AF_INET = 2;
+    AF_INET6 = 10;
+  }
+  // These are 8 bit unsigned.
+  optional AddressFamily family  = 1;
+
+  enum TCPState {  //  from tcp_states.h
+    ESTABLISHED = 1;
+    SYN_SENT    = 2;
+    SYN_RECV    = 3;
+    FIN_WAIT1   = 4;
+    FIN_WAIT2   = 5;
+    TIME_WAIT   = 6;
+    CLOSE       = 7;
+    CLOSE_WAIT  = 8;
+    LAST_ACK    = 9;
+    LISTEN      = 10;
+    CLOSING     = 11;    /* Now a valid state */
+  }
+  optional TCPState state = 2;  // TODO - is this the correct enum?
+  optional int32 timer    = 3;
+  optional int32 retrans  = 4;
+
+  optional InetSocketIDProto sock_id = 5;
+  optional int32 expires =  6;
+  optional int32 rqueue  =  7;
+  optional int32 wqueue  =  8;
+  optional int32 uid     =  9;
+  optional int32 inode   = 10;
+}
+
+// Proto representation for struct tcpvegas_info.
+message TCPVegasInfoProto {
```

Comment:
I don't anticipate us ever needing this message type. We have no plans to ever deploy TCP-Vegas